nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/AbstractSNMPProcessor.java

```diff
@@ -16,413 +16,249 @@
-            .required(false)
+            .displayName("SNMP Security Name / Username for SNMP v3 Authentication")
+            .description("Security name / User name used for SNMP exchanges.")
+            .required(true)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V3)
             .build();
 
-    /** property to define SNMP authentication protocol to use */
-    public static final PropertyDescriptor SNMP_AUTH_PROTOCOL = new PropertyDescriptor.Builder()
-            .name("snmp-authentication-protocol")
-            .displayName("SNMP Authentication Protocol")
-            .description("SNMP Authentication Protocol to use")
+    public static final PropertyDescriptor SNMP_PRIVACY_PROTOCOL = new PropertyDescriptor.Builder()
+            .name("snmp-private-protocol")
+            .displayName("SNMP Privacy Protocol")
+            .description("Privacy allows for encryption of SNMP v3 messages to ensure confidentiality of data.")
             .required(true)
-            .allowableValues("MD5", "SHA", "")
+            .allowableValues(DES, DES3, AES128, AES192, AES256, NO_ENCRYPTION)
             .defaultValue("")
+            .dependsOn(SNMP_SECURITY_LEVEL, AUTH_PRIV)
             .build();
 
-    /** property to define SNMP authentication password to use */
-    public static final PropertyDescriptor SNMP_AUTH_PASSWORD = new PropertyDescriptor.Builder()
-            .name("snmp-authentication-passphrase")
-            .displayName("SNMP Authentication pass phrase")
-            .description("Pass phrase used for SNMP authentication protocol")
-            .required(false)
+    public static final PropertyDescriptor SNMP_PRIVACY_PASSWORD = new PropertyDescriptor.Builder()
+            .name("snmp-private-protocol-passphrase")
+            .displayName("SNMP Privacy Protocol Passphrase")
+            .description("Pass phrase used for SNMP privacy protocol")
+            .required(true)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .sensitive(true)
+            .dependsOn(SNMP_SECURITY_LEVEL, AUTH_PRIV)
             .build();
 
-    /** property to define SNMP private protocol to use */
-    public static final PropertyDescriptor SNMP_PRIV_PROTOCOL = new PropertyDescriptor.Builder()
-            .name("snmp-private-protocol")
-            .displayName("SNMP Private Protocol")
-            .description("SNMP Private Protocol to use")
+    public static final PropertyDescriptor SNMP_AUTH_PROTOCOL = new PropertyDescriptor.Builder()
+            .name("snmp-authentication-protocol")
+            .displayName("SNMP Authentication Protocol")
+            .description("Hash based auhentication protocols for secure authentication.")
             .required(true)
-            .allowableValues("DES", "3DES", "AES128", "AES192", "AES256", "")
+            .allowableValues(MD5, SHA, NO_AUTHENTICATION)
```

Comment:
`None` option seems to be obsolete because this property depends on Security Level and when the property is visible, it is mandatory.